csharp/delegates-and-events/FileLogger.cs

```diff
@@ -3,34 +3,38 @@
 
 namespace DelegatesAndEvents
 {
-public class FileLogger
-{
-    private readonly string logPath;
-    public FileLogger(string path)
-    {
-        logPath = path;
-        Logger.WriteMessage += LogMessage;
-    }
-    
-    public void DetachLog() => Logger.WriteMessage -= LogMessage;
-    // make sure this can't throw.
-    private void LogMessage(string msg)
+    // <SnippetFileLogger>
+    public class FileLogger
     {
-        try {
-            using (var log = File.AppendText(logPath))
+        private readonly string logPath;
+        public FileLogger(string path)
+        {
+            logPath = path;
+            Logger.WriteMessage += LogMessage;
+        }
+
+        public void DetachLog() => Logger.WriteMessage -= LogMessage;
+        // make sure this can't throw.
+        private void LogMessage(string msg)
+        {
+            try
             {
-                log.WriteLine(msg);
-                log.Flush();
+                using (var log = File.AppendText(logPath))
+                {
+                    log.WriteLine(msg);
+                    log.Flush();
+                }
+            }
+            catch (Exception)
+            {
+                // Hmm. We caught an exception while 
```

Comment:
nit: since comments aren't localized, this might be better if it were in the example explanation rather than an inline comment.

Reply:
There is explanatory text in the article as well. I'll leave the comment for those that just download the code.